cognite/neat/rules/exporter/dm/rules2dms.py

```diff
@@ -1,2 +1,315 @@
+                    description=property_definition.description,
+                )
+            else:
+                logging.warning(_exceptions.Warning61(property_id).message)
+                warnings.warn(
+                    _exceptions.Warning61(property_id).message,
+                    category=_exceptions.Warning61,
+                    stacklevel=2,
+                )
+
+        return view_properties
+
+    def to_cdf(self, client: CogniteClient):
+        """Creates the data model in CDF.
+
+        Args:
+            client (CogniteClient): Cognite client.
+        """
+        self.create_space(client)
+        self.create_containers(client)
+        self.create_views(client)
+        self.create_data_model(client)
+
+    def create_space(self, client: CogniteClient):
+        if not client.data_modeling.spaces.retrieve(space=self.space):
+            logging.info(f"Creating space {self.space}")
+            _ = client.data_modeling.spaces.apply(SpaceApply(space=self.space))
+        else:
+            logging.info(f"Space {self.space} already exists. Skipping creation!")
+
+    def create_containers(self, client: CogniteClient):
+        cdf_containers = {}
+        if containers := client.data_modeling.containers.list(space=self.space, limit=-1):
+            cdf_containers = {container.external_id: container for container in containers}
+
+        if existing_containers := set(self.containers.keys()).intersection(set(cdf_containers.keys())):
+            logging.warning(_exceptions.Warning62(existing_containers, self.space).message)
+            warnings.warn(
+                _exceptions.Warning62(existing_containers, self.space).message,
+                category=_exceptions.Warning62,
+                stacklevel=2,
+            )
+        else:
+            for container_id in self.containers:
+                logging.info(f"Creating container {container_id} in space {self.space}")
+                _ = client.data_modeling.containers.apply(self.containers[container_id])
+
+    def create_views(self, client: CogniteClient, force: bool = False):
+        cdf_views = {}
+        if views := client.data_modeling.views.list(space=self.space, limit=-1):
+            cdf_views = {container.external_id: container for container in views}
+
+        if existing_views := set(self.views.keys()).intersection(set(cdf_views.keys())):
+            logging.warning(_exceptions.Warning63(existing_views, self.version, self.space).message)
+            warnings.warn(
+                _exceptions.Warning63(existing_views, self.version, self.space).message,
+                category=_exceptions.Warning63,
+                stacklevel=2,
+            )
+        else:
+            for view_id in self.views:
+                logging.info(f"Creating view {view_id} version {self.views[view_id].version} in space {self.space}")
+                _ = client.data_modeling.views.apply(self.views[view_id])
```

Comment:
This looks strange to me. Why are you not creating the views when you get a warning?

Reply:
I moved test if any of data model components exsist and if they do... then raise `Error60`